### ISSUE TITLE
SyncFilter should build from certain tags instead of hardcoded

### ISSUE THREAD

**Issue by danielo515** [Jermolene/TiddlyWiki5#1746]
Hello, the tiddler $:/config/SyncFilter is a very useful feature. The only problem is that you have to override it to add values to it. This is not a problem if, for example, only one plugin does, but if there are some plugins or customizations that wants to blacklist certain tiddlers we have a problem.

In my opinion, this filter should be "composed" by fetching all the tiddlers with the tag $:/tags/SyncFilter. That way we can blacklist tiddlers without creating conflicts with other modlues and/or plugins

This may solve our discussion at #1744

**Comment by pmario** [comment]
You can change this filter in a way that does exactly that. So I don't see a problem?

**Comment by danielo515** [comment]
I don't understand your point @pmario , could you please elaborate a bit?
Thanks

**Comment by pmario** [comment]
$:/config/SyncFilter contains: 

`[is[tiddler]] -$:/HistoryList -$:/Import -$:/isEncrypted -[prefix[$:/status]] -[prefix[$:/state]] -[prefix[$:/temp]]`

if you add `-[tag[$:/tags/excludeSync]]` it will exclude all tiddlers tagged $:/tags/excludeSync. That's what you want. So your plugins could just add this parameter, if not there, and use it. ... done. ... If it turns out to be useful, you can create a pull request.

**Comment by danielo515** [comment]
No that is not exactly what I want. I want to add items to that filter by tagging tiddlers that contain that rules. For example, I have a tiddler that contains 

```
-[[example]] -[tag[config]] 
```

I want to tag that tiddler with the appropriate tag and have its text added to the exclude filter. I don't want to tag each particular tiddler. This will allow plugin wide configurations with just one tiddler. What you are suggesting is more useful from a end user point of view. 

And what I want the most is to not have to modify any shadow tiddler. You suggested that my plugin should modify the shadow tiddler. What if the user does the same? What if another plugin needs a similar functionality?

**Comment by Jermolene** [comment]
Hi @danielo515 I don't think that it is feasible to have the sync filter built up in pieces. There are two issues:

* The filter syntax isn't really suitable for building up by concatenation. The position of each portion would affect its interpretation. So, if one plugin adds "[[$:/StoryList]]" and another adds "-[[$:/StoryList]]" then the result will depend upon the ordering
* Even if that were fixed, the specification of the sync filter isn't really something that can be done at the level of a plugin. In complicated cases with multiple sync adaptors it would be hard for each plugin to know enough about the system as a whole to choose the correct configuration

My belief is that there are certain bits of configuration that are inherently "singular", such as `$:/DefaultTiddlers`. These things need to be configured at the level of an edition (a packaged distribution of TiddlyWiki adapted for a specific purpose/configuration); plugins should focus on figuring out sensible defaults.

**Comment by danielo515** [comment]
Hello @Jermolene 

If you see my proposal very clumsy I think the best approach is what @pmario have suggested. What do you think about that? In my opinion is important to have the opportunity to decide what yo don't want to sync.

**Comment by Jermolene** [comment]
@pmario's proposal will be a bit brittle in this situation since I think it requires a tag to be placed on the `$:/StoryList` tiddler.

**Comment by danielo515** [comment]
Yes, it requires a tag on that tiddler. But since maybe is not a good idea for that particular tiddler, it can be useful for many other tiddlers. At the end it gives more control to the user to decide what they want to sync.

**Comment by tobibeer** [comment]
Instead of tagging "$:/StoryList", there can be more sophisticated namespaces, from which to remove the prefix before processing, e.g.: `[removeprefix[$:/config/DoNotSync/]` and then having `$:/config/DoNotSync/$:/StoryList`.

**Comment by Jermolene** [comment]
As I say above, I don't think that plugins should be trying to modify the sync filter. The mechanism @tobibeer suggests is better, but I'm not convinced that it's needed (the motivation for the proposal is I think an attempt to fix a much deeper problem with managing per-user state data in multi-user configurations).
